src/storage/write_ahead_log/disk_log_consumer_task.cpp

```diff
@@ -81,8 +86,12 @@ void DiskLogConsumerTask::DiskLogConsumerTaskLoop() {
       // 2) There is a filled buffer to write to the disk
       // 3) LogManager has shut down the task
       // 4) Our persist interval timed out
-      disk_log_writer_thread_cv_.wait_for(lock, persist_interval_,
-                                          [&] { return do_persist_ || !filled_buffer_queue_->Empty() || !run_task_; });
+      next_sleep =
+          std::min(disk_log_writer_thread_cv_.wait_for(
+                       lock, curr_sleep, [&] { return do_persist_ || !filled_buffer_queue_->Empty() || !run_task_; })
+                       ? persist_interval_
+                       : curr_sleep * 2,
+                   max_sleep);
```

Comment:
I would suggest breaking this up into two statements. This is a bit complicated and difficult to parse.

Reply:
1 upped you: 3 lines